libqltipc/ql-tipc/keymaster_serializable.c

```diff
@@ -76,6 +76,25 @@ int km_boot_params_serialize(const struct km_boot_params *params, uint8_t** out,
     return TRUSTY_ERR_NONE;
 }
 
+int km_boot_patchlevel_serialize(const struct km_boot_patchlevel *params, uint8_t** out,
+                             uint32_t *out_size)
+{
+    uint8_t *tmp;
+
+    if (!out || !params || !out_size) {
+        return TRUSTY_ERR_INVALID_ARGS;
+    }
+    *out_size = (sizeof(params->boot_patchlevel));
+    *out = trusty_calloc(*out_size, 1);
+    if (!*out) {
+        return TRUSTY_ERR_NO_MEMORY;
+    }
+
+    tmp = append_uint32_to_buf(*out, params->boot_patchlevel);
```

Comment:
tmp is not used in the code

Reply:
> Pass boot_patchlevel to trusty.
> 
> Boot_patchlevel specifies the kernel image security patch level that must be installed on the device for a key to be used. The value appears in the form YYYYMMDD, representing the date of the system security patch. For example, if a key were generated on an Android device with the system's Oct 1, 2010 security patch installed, this value would be 20101001.
> 
> Tracked-On: OAM-99223 Signed-off-by: yuxincui [yuxin.cui@intel.com](mailto:yuxin.cui@intel.com)

you can change the commit msg as:
" Pass boot_patchlevel to trusty via KM_CONFIGURE_BOOT_PATCHLEVEL"